make in clean repo fails on macOS

If I clone the repo and run a bare "make", I get an error:

```
❯ make
go: downloading github.com/guregu/null v4.0.0+incompatible
go: downloading github.com/golang-migrate/migrate/v4 v4.16.2
go: downloading github.com/iancoleman/strcase v0.2.0
go: downloading github.com/jackc/pgerrcode v0.0.0-20220416144525-469b46aa5efa
go: downloading github.com/jackc/pgx/v5 v5.3.1
go: downloading golang.org/x/exp v0.0.0-20230713183714-613f0c0eb8a1
go: downloading github.com/jsuar/go-cron-descriptor v0.1.0
go: downloading github.com/robfig/cron v1.2.0
go: downloading github.com/hibiken/asynq v0.24.0
go: downloading github.com/lib/pq v1.10.2
go: downloading go.uber.org/atomic v1.10.0
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/jackc/pgpassfile v1.0.0
go: downloading github.com/jackc/pgservicefile v0.0.0-20221227161230-091c0ba34f0a
go: downloading github.com/jackc/puddle/v2 v2.2.0
go: downloading golang.org/x/crypto v0.7.0
go: downloading golang.org/x/text v0.9.0
go: downloading go.uber.org/zap v1.24.0
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading golang.org/x/sync v0.2.0
go: downloading go.uber.org/multierr v1.9.0
go: downloading github.com/go-redis/redis/v8 v8.11.2
go: downloading github.com/google/uuid v1.3.0
go: downloading google.golang.org/protobuf v1.28.1
go: downloading golang.org/x/sys v0.8.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/robfig/cron/v3 v3.0.1
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/spf13/cast v1.3.1
go: downloading github.com/dgryski/go-rendezvous v0.0.0-20200823014737-9f7001d12a5f
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/pkg/errors v0.9.1
package github.com/acaloiaro/neoq_test
	imports github.com/acaloiaro/neoq/testutils: build constraints exclude all Go files in /tmp/neoq/testutils
package github.com/acaloiaro/neoq/backends/postgres_test
	imports github.com/acaloiaro/neoq/testutils: build constraints exclude all Go files in /tmp/neoq/testutils
package github.com/acaloiaro/neoq/backends/postgres_test
	imports github.com/acaloiaro/neoq/testutils: build constraints exclude all Go files in /tmp/neoq/testutils
# github.com/acaloiaro/neoq/backends/memory_test
vet: backends/memory/memory_backend_test.go:172:24: undefined: memory.TestingBackend
Makefile:16: *** [vet] error 1

#0  vet at /private/tmp/neoq/Makefile:16
#1  all at /private/tmp/neoq/Makefile:3
```

I'm on macOS 13.4.1.
```
❯ sw_vers 
ProductName:		macOS
ProductVersion:		13.4.1
ProductVersionExtra:	(c)
BuildVersion:		22F770820d
```

--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ fmt:
 
 .PHONY: vet
 vet:
-	@go vet ./...
+	@go vet -tags testing ./...
 
 .PHONY: test
 test: install-gotestsum

